raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -942,6 +958,10 @@ private FetchResponseData tryCompleteFetchRequest(
      * to in case it is not.
      */
     private Optional<OffsetAndEpoch> validateFetchOffsetAndEpoch(long fetchOffset, int lastFetchedEpoch) {
+        if (fetchOffset == 0 && lastFetchedEpoch == 0) {
```

Comment:
Is this a bug fix or an optimization?

Reply:
I guess I'd call it a bug fix. I ran into an error because it was not handled consistently by both implementations. I thought in the end that it might be safer to make this a special case.